### DISCUSSION TITLE
Having a problem with export_pseudobulk

### DISCUSSION THREAD

**Discussion by kae-rizz** [aertslab/scenicplus#138] [Q&A]
Hi,

I am having a problem with my files finishing the export_pseudobulk process.  They all begin, but some do not say done, and then no bed files are created.  1-11 finished, but 12,13,14, and 15 do not. Because they do not finishing in export_pseudobulk this throws an error when running peak_calling because there is no bed file.  However, no error is thrown when running export_pseudobulk.  For reference, I am using the pbmc_multiome notebook tutorial.

Below I put the output I get when running export_pseudobulk:

2023-04-04 14:55:01,346	INFO worker.py:1544 -- Started a local Ray instance. View the dashboard at http://127.0.0.1:826
(export_pseudobulk_ray pid=2954219) 2023-04-04 14:55:04,041 cisTopic     INFO     Creating pseudobulk for 1
(export_pseudobulk_ray pid=2954213) 2023-04-04 14:55:04,052 cisTopic     INFO     Creating pseudobulk for 0
(export_pseudobulk_ray pid=2954219) 2023-04-04 14:55:04,296 cisTopic     INFO     1 done!
(export_pseudobulk_ray pid=2954212) 2023-04-04 14:55:04,223 cisTopic     INFO     Creating pseudobulk for 10
(export_pseudobulk_ray pid=2954211) 2023-04-04 14:55:04,289 cisTopic     INFO     Creating pseudobulk for 13
(export_pseudobulk_ray pid=2954219) 2023-04-04 14:55:04,341 cisTopic     INFO     Creating pseudobulk for 2
(export_pseudobulk_ray pid=2954213) 2023-04-04 14:55:04,408 cisTopic     INFO     0 done!
(export_pseudobulk_ray pid=2954214) 2023-04-04 14:55:04,408 cisTopic     INFO     Creating pseudobulk for 11
(export_pseudobulk_ray pid=2954213) 2023-04-04 14:55:04,452 cisTopic     INFO     Creating pseudobulk for 3
(export_pseudobulk_ray pid=2954212) 2023-04-04 14:55:04,458 cisTopic     INFO     10 done!
(export_pseudobulk_ray pid=2954215) 2023-04-04 14:55:04,414 cisTopic     INFO     Creating pseudobulk for 12
(export_pseudobulk_ray pid=2954217) 2023-04-04 14:55:04,468 cisTopic     INFO     Creating pseudobulk for 14
(export_pseudobulk_ray pid=2954212) 2023-04-04 14:55:04,514 cisTopic     INFO     Creating pseudobulk for 4
(export_pseudobulk_ray pid=2954214) 2023-04-04 14:55:04,598 cisTopic     INFO     11 done!
(export_pseudobulk_ray pid=2954215) 2023-04-04 14:55:04,573 cisTopic     INFO     12 done!
(export_pseudobulk_ray pid=2954211) 2023-04-04 14:55:04,531 cisTopic     INFO     Creating pseudobulk for 5
(export_pseudobulk_ray pid=2954214) 2023-04-04 14:55:04,643 cisTopic     INFO     Creating pseudobulk for 8
(export_pseudobulk_ray pid=2954215) 2023-04-04 14:55:04,621 cisTopic     INFO     Creating pseudobulk for 6
(export_pseudobulk_ray pid=2954217) 2023-04-04 14:55:04,650 cisTopic     INFO     Creating pseudobulk for 7
(export_pseudobulk_ray pid=2954212) 2023-04-04 14:55:04,740 cisTopic     INFO     4 done!
(export_pseudobulk_ray pid=2954212) 2023-04-04 14:55:04,782 cisTopic     INFO     Creating pseudobulk for 9
(export_pseudobulk_ray pid=2954211) 2023-04-04 14:55:04,742 cisTopic     INFO     5 done!
(export_pseudobulk_ray pid=2954216) 2023-04-04 14:55:04,762 cisTopic     INFO     Creating pseudobulk for 15
(export_pseudobulk_ray pid=2954219) 2023-04-04 14:55:04,907 cisTopic     INFO     2 done!
(export_pseudobulk_ray pid=2954213) 2023-04-04 14:55:04,860 cisTopic     INFO     3 done!
(export_pseudobulk_ray pid=2954215) 2023-04-04 14:55:04,850 cisTopic     INFO     6 done!

**Comment by SeppeDeWinter** [comment]
Hi @kae-rizz 

Sometimes this is caused by having spaces (or other special characters) in the names. For you it seems to be cluster numbers so I guess this is not the problem for you?

If you run the same code but with a single core it will also give a better error message (ray does not pass the actual error), this can also help with troubleshooting.

Best,

Seppe
